server/models/messageModel.js

```diff
@@ -0,0 +1,29 @@
+const mongoose = require("mongoose");
+
+const messageSchema = new mongoose.Schema({
+  messages: [
+    {
+      sender: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Profile",
+      },
+      content: {
+        type: String,
+      },
+      timeCreated: {
+        type: Date,
+      },
+      wasRead: {
+        type: Boolean,
+      },
+      chatId: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "Chat",
+      },
+    },
+  ],
```

Comment:
Would it be better if the message schema was for a single message. The Chat Model's messages field can then be an array of object ids referencing the message model. If there are a lot of messages between users, there is a risk that the size of the array will exceed the maximum document size allowed by MongoDB. Storing only the objectIds in an array rather than the entire message information may reduce that size.

Reply:
I agree with that approach, it should also make your logic easier for finding the most recent message because I believe there's a way to make findOne in mongoose return the most recent matching document (or it does it automatically)